src/template_parser.rs

```diff
@@ -333,22 +333,27 @@ fn parse_boolean_method<'a>(
 
 fn parse_commit_or_change_id_method<'a>(
     name: Pair<Rule>,
-    _args: Pairs<Rule>,
+    args: Pairs<Rule>,
 ) -> Result<Property<'a, CommitOrChangeId<'a>>, TemplateParseError> {
     fn wrap_fn<'a, O>(
         f: impl Fn(&CommitOrChangeId<'a>) -> O + 'a,
     ) -> Box<dyn TemplateProperty<CommitOrChangeId<'a>, Output = O> + 'a> {
         Box::new(TemplatePropertyFn(f))
     }
+    let args = args.as_str();
     // TODO: validate arguments
+    let int_arg: Option<usize> = (!args.is_empty()).then(|| {
```

Comment:
@yuja What is the proper way of doing this?

Reply:
We'll need a parsing (tokenizing) rule for integer literal, `Property::Integer` type, and some way to evaluate arguments against the top-level context (i.e. `&Commit` instead of the method receiver.) The first two are easy, the last one will require some consideration.

If you can wait a couple of days, I'll take a look. I think I can at least add an integer literal support (which doesn't require any context.)

Reply:
WIP: https://github.com/martinvonz/jj/compare/main...yuja:jj:push-0f24397482f9

I'll send the remainder if #1191 gets merged.

Example of method arguments handling:
https://github.com/martinvonz/jj/commit/0e7180c5e7ad93c8d007a735c3799d1f9661a24d

You'll need to add `.try_into_integer()` to coerce to `i64` property (see `.try_into_boolean()`), and turn it to `usize` (ignoring negative number) on evaluation phase.

Reply:
Thanks, Yuya!

I would find it simpler to just pass the `String` to the `TemplatePropertyFn`  and have it do the parsing of the integer. This might mean that `TemplateProperty`-es would need to be able to return an error. I guess you're planning for a time where the argument of a method could be another expression in the templating language.